DuckDuckGo/SettingsViewModel.swift

```diff
@@ -0,0 +1,391 @@
+            longPressPreviews: appSettings.longPressPreviews,
+            allowUniversalLinks: appSettings.allowUniversalLinks,
+            activeWebsiteAccount: nil, // Default value for logins
+            netPSubtitle: "", // Default value for Network Protection
+            version: versionProvider.versionAndBuildNumber
+        )
+        setupSubscribers()
+    }
+        
+    private func firePixel(_ event: Pixel.Event) {
+        Pixel.fire(pixel: event)
+    }
+    
+    private func enableVoiceSearch(completion: @escaping (Bool) -> Void) {
+        SpeechRecognizer.requestMicAccess { permission in
+            if !permission {
+                completion(false)
+                return
+            }
+            AppDependencyProvider.shared.voiceSearchHelper.enableVoiceSearch(true)
+            completion(true)
+        }
+    }
+    
+#if NETWORK_PROTECTION
+    private func updateNetPCellSubtitle(connectionStatus: ConnectionStatus) {
+        switch NetworkProtectionAccessController().networkProtectionAccessType() {
+        case .none, .waitlistAvailable, .waitlistJoined, .waitlistInvitedPendingTermsAcceptance:
+            self.state.netPSubtitle = VPNWaitlist.shared.settingsSubtitle
+        case .waitlistInvited, .inviteCodeInvited:
+            switch connectionStatus {
+            case .connected:
+                self.state.netPSubtitle = UserText.netPCellConnected
+            default:
+                self.state.netPSubtitle = UserText.netPCellDisconnected
+            }
+        }
+    }
+#endif
+}
+
+// MARK: Subscribers
+extension SettingsViewModel {
+    
+    private func setupSubscribers() {
+               
+
+#if NETWORK_PROTECTION
+        connectionObserver.publisher
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] status in
+                self?.updateNetPCellSubtitle(connectionStatus: status)
+            }
+            .store(in: &cancellables)
+#endif
+        
+    }
+}
+
+// MARK: Public Methods
+extension SettingsViewModel {
+    
+    func onAppear() {
+        initState()
+    }
+    
+    func setAsDefaultBrowser() {
+        firePixel(.defaultBrowserButtonPressedSettings)
+        guard let url = URL(string: UIApplication.openSettingsURLString) else { return }
+        UIApplication.shared.open(url)
+    }
+    
+    @MainActor func shouldPresentLoginsViewWithAccount(accountDetails: SecureVaultModels.WebsiteAccount) {
+        state.activeWebsiteAccount = accountDetails
+        presentLegacyView(.logins)
+    }
+    
+    func openEmailProtection() {
+        UIApplication.shared.open(URL.emailProtectionQuickLink,
+                                  options: [:],
+                                  completionHandler: nil)
+    }
+        
+    @MainActor func openCookiePopupManagement() {
+        pushViewController(legacyViewProvider.autoConsent)
+    }
```

Comment:
This could be replaced by `presentLegacyView(.autoconsent)`, right?

Reply:
No, it can't because PushViewcontroller is generic and expects a UIViewController.